iXBRLViewerPlugin/viewer/src/js/textblockviewer.js

```diff
@@ -0,0 +1,36 @@
+// Copyright 2023 Workiva Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import $ from 'jquery';
```

Comment:
I hadn't planned to eliminate it entirely, just to stop using it for things where there's a better ES6 alternative (particularly `$.each()` and `$.map()`).

I think it would be quite hard to rewrite all of the jquery node navigation we use in the viewer, but I've never really looked into it.

Reply:
Having done some googling, this actually looks less hard than I imagined, but obviously quite a lot of work to eliminate it entirely.

There's various places in the viewer where we flip back and forth between JS and jQuery nodes, and it probably makes sense to switch those to pure JS first, particularly the performance critical pre-processing code in `viewer.js`.